### ISSUE TITLE
Inventory error?! Unknown signal. Trying to add inventory to yeasts.

### ISSUE THREAD

**Issue by mattiasmaahl** [Brewtarget/brewtarget#601]
I'm trying to add inventory to the Yeasts, one of the included ones (Danstar - Nottingham).
But Brewtarget crashes abruptly with the following error:
![bild](https://user-images.githubusercontent.com/1370239/123970551-df1bf900-d9c1-11eb-8801-a633e7ecb8c0.png)

 
![bild](https://user-images.githubusercontent.com/1370239/123970710-07a3f300-d9c2-11eb-97d6-1dfda1480634.png)

can anyone shed some light on this as I don't fully follow the redo/undo methods?

**Comment by mattiasmaahl** [comment]
So far I'm having problems with Yeast, Hops and Misc. all with the above error.
Fermentables was OK to add to inventory for some reason.

This could be because of my database as well, have to do more testing tomorrow.

**Comment by matty0ung** [comment]
I can reproduce this.  Will have a look.

**Comment by matty0ung** [comment]
OK, I have a fix for this.  Will backport it to Brewtarget and submit in the next day or so.  (Need a couple of tweaks for the backport as `NamedEntityWithInventory` doesn't yet exist in BT :o>)

**Comment by mattiasmaahl** [comment]
Nice, I'll continue my work with inventory printout using fermentables for now then awaiting your fix. 😊

**Comment by matty0ung** [comment]
The crash was caused by an invalid property name in YeastTabelModel.cpp (and equivalent places for other things with inventory).  I've changed the hard-coded string property names for symbolic ones (so we should get compiler errors in the future for bad property names), as has already been done in most other parts of the code.  I also pulled out the common inventory functions from Yeast, Hop, etc to an intermediate parent class (NamedEntityWithInventory).  So the crashing seems to stop now...

However, there is still an issue that we're not always setting the inventory ID, which means inventory amounts aren't getting stored.  There may be some small fix in Brewtarget that addresses this, and I'm just not instantly seeing it.  Otherwise, when I bring the new DB layer back, there's new code (in a new Inventory class) that does all the right things.
